### ISSUE TITLE
VidCoder multiple crashes

### ISSUE THREAD

**Issue by gstock99** [RandomEngy/VidCoder#206]
When ever I try to open any kind of dialog box, whether it be a import / export, File Open, etc.  The program crashes.  Here is the data I retrieved from Event Viewer.  If there is further info required, let me know and I'll try to do it.  I don't think my problems is a VidCoder issue, but more of a Windows 10 issue.  I could use some help solving this.

It works fine on my desktop, but not my laptop (both Windows 10).  I am enclosing both the portable and full installed version information.

Below are the copied info from Windows Event Viewer application logs:

# **Vidcoder version:  Portable Version 2.52**
## **Source:  .NET Runtime**
Log Name:      Application
Source:        .NET Runtime
Date:          5/30/2017 1:37:20 PM
Event ID:      1026
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      Greg_FCTC-PC
Description:
Application: VidCoder.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: exception code c0000005, exception address 00007FFB2C379E08
Stack:

Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Level>2</Level>
    <Task>0</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-05-30T17:37:20.161515300Z" />
    <EventRecordID>2549</EventRecordID>
    <Channel>Application</Channel>
    <Computer>Greg_FCTC-PC</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Application: VidCoder.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: exception code c0000005, exception address 00007FFB2C379E08
Stack:
</Data>
  </EventData>
</Event>

## **Source:  Application Error**
Log Name:      Application
Source:        Application Error
Date:          5/30/2017 1:37:20 PM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      xxxxxxxxx
Description:
Faulting application name: VidCoder.exe, version: 2.52.0.0, time stamp: 0x5921dd8a
Faulting module name: KERNELBASE.dll, version: 10.0.15063.296, time stamp: 0xa0527b0c
Exception code: 0xc0000005
Fault offset: 0x0000000000069e08
Faulting process id: 0x201c
Faulting application start time: 0x01d2d96b5e06220f
Faulting application path: C:\Users\xxxxxxxxx\AppData\Local\Temp\RarSFX0\VidCoder.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 4dbf5b0b-4176-474f-84ca-0a313ef7292e
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-05-30T17:37:20.230256700Z" />
    <EventRecordID>2550</EventRecordID>
    <Channel>Application</Channel>
    <Computer>xxxxxxxxx</Computer>
    <Security />
  </System>
  <EventData>
    <Data>VidCoder.exe</Data>
    <Data>2.52.0.0</Data>
    <Data>5921dd8a</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>10.0.15063.296</Data>
    <Data>a0527b0c</Data>
    <Data>c0000005</Data>
    <Data>0000000000069e08</Data>
    <Data>201c</Data>
    <Data>01d2d96b5e06220f</Data>
    <Data>C:\Users\xxxxxxxxx\AppData\Local\Temp\RarSFX0\VidCoder.exe</Data>
    <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data>
    <Data>4dbf5b0b-4176-474f-84ca-0a313ef7292e</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

# **Vidcoder version:  Full Version 2.52**
**## Source:  .NET Runtime**
Log Name:      Application
Source:        Application Error
Date:          5/30/2017 6:55:11 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      xxxxxxxxx
Description:
Faulting application name: VidCoder.exe, version: 2.52.0.0, time stamp: 0x5921dd8a
Faulting module name: KERNELBASE.dll, version: 10.0.15063.296, time stamp: 0xa0527b0c
Exception code: 0xc0000005
Fault offset: 0x0000000000069e08
Faulting process id: 0x3a7c
Faulting application start time: 0x01d2d93329d97207
Faulting application path: D:\Program Files\VidCoder\VidCoder.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 2b028577-00bb-40db-8ce4-4064745c4447
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-05-30T10:55:11.460476100Z" />
    <EventRecordID>2354</EventRecordID>
    <Channel>Application</Channel>
    <Computer>xxxxxxxxx</Computer>
    <Security />
  </System>
  <EventData>
    <Data>VidCoder.exe</Data>
    <Data>2.52.0.0</Data>
    <Data>5921dd8a</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>10.0.15063.296</Data>
    <Data>a0527b0c</Data>
    <Data>c0000005</Data>
    <Data>0000000000069e08</Data>
    <Data>3a7c</Data>
    <Data>01d2d93329d97207</Data>
    <Data>D:\Program Files\VidCoder\VidCoder.exe</Data>
    <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data>
    <Data>2b028577-00bb-40db-8ce4-4064745c4447</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

## **Source:  Application Error**
Log Name:      Application
Source:        Application Error
Date:          5/30/2017 6:55:11 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      xxxxxxxxx
Description:
Faulting application name: VidCoder.exe, version: 2.52.0.0, time stamp: 0x5921dd8a
Faulting module name: KERNELBASE.dll, version: 10.0.15063.296, time stamp: 0xa0527b0c
Exception code: 0xc0000005
Fault offset: 0x0000000000069e08
Faulting process id: 0x3a7c
Faulting application start time: 0x01d2d93329d97207
Faulting application path: D:\Program Files\VidCoder\VidCoder.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 2b028577-00bb-40db-8ce4-4064745c4447
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-05-30T10:55:11.460476100Z" />
    <EventRecordID>2354</EventRecordID>
    <Channel>Application</Channel>
    <Computer>xxxxxxxxx</Computer>
    <Security />
  </System>
  <EventData>
    <Data>VidCoder.exe</Data>
    <Data>2.52.0.0</Data>
    <Data>5921dd8a</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>10.0.15063.296</Data>
    <Data>a0527b0c</Data>
    <Data>c0000005</Data>
    <Data>0000000000069e08</Data>
    <Data>3a7c</Data>
    <Data>01d2d93329d97207</Data>
    <Data>D:\Program Files\VidCoder\VidCoder.exe</Data>
    <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data>
    <Data>2b028577-00bb-40db-8ce4-4064745c4447</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

**Comment by RandomEngy** [comment]
Looks like something is corrupted in your Windows or .NET install. `KERNELBASE.dll` is a system DLL.

**Comment by dot256** [comment]
very frequent crashes:

Faulting application name: VidCoder.exe, version: 2.52.0.0, time stamp: 0x5921dd8a
Faulting module name: KERNELBASE.dll, version: 10.0.15063.296, time stamp: 0xa0527b0c
Exception code: 0xe0434352
Fault offset: 0x0000000000069e08
Faulting process id: 0x2500
Faulting application start time: 0x01d2d96a1bff548a
Faulting application path: D:\data\soft\media\video encode\vidcoder\RarSFX0\VidCoder.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 4c1d7871-12e9-48e3-858c-13a2899adfd5
Faulting package full name: 
Faulting package-relative application ID:

**Comment by RandomEngy** [comment]
Ahh now this is some useful information. Will take a look at it.

**Comment by gstock99** [comment]
OK, I'm sorry if this is not the information you are looking for, but I can tell you this.  I installed the 32-bit version and I able to import / export, File Open and all other windows type dialog boxes.  If I then open the portable version of the 64-bit program, if I had a queue loaded in the 32-bit version it transfers in and I am able to use to run my queue.  With H265 there is a significant speed advantage of H264 with 32-bit.

I am not saying its necessarily a VidCoder issue, but if anyone as any ideas with how to resolve this.  I've tried everything short of re-installing windows (Not really an option since its a work laptop and I would have to jump thru hoops to re-license proprietary work related software).

Thanks,
Greg

PC Details:
Windows 10 Pro - ver 1703
Dell Precision M4800
i7-4810MQ
8GB RAM
AMD FirePro M5100
480GB SSD

**Comment by RandomEngy** [comment]
@dot256 Do you have a custom value for SettingsDirectory in your app config file? If so, what is it?

**Comment by dot256** [comment]
VidCoderWorker.exe.config

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
	<appSettings>
		<!-- Uncomment this line and set the value if you want VidCoder to look in a specific directory for app settings instead of %appdata%\VidCoder
				 Use "." to store the settings files in the same folder as VidCoder.exe.
		     You must also change the same value in VidCoder.exe.config -->
		<add key="SettingsDirectory" value="." />
	</appSettings>
	
	<startup useLegacyV2RuntimeActivationPolicy="true">
		<supportedRuntime version="v4.0"/>
	</startup>
</configuration>
```

VidCoder.exe.config

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<appSettings>
		<!-- Uncomment this line and set the value if you want VidCoder to look in a specific directory for app settings instead of %appdata%\VidCoder
				 Use "." to store the settings files in the same folder as VidCoder.exe.
		     You must also change the same value in VidCoderWorker.exe.config -->
		<add key="SettingsDirectory" value="." />
	</appSettings>

  <startup useLegacyV2RuntimeActivationPolicy="true">
    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6" />
  </startup>

  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Practices.ServiceLocation" publicKeyToken="31bf3856ad364e35" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-1.3.0.0" newVersion="1.3.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-8.0.0.0" newVersion="8.0.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

**Comment by RandomEngy** [comment]
Aha. Just as I suspected. I'll have this fixed in the next version.

**Comment by dot256** [comment]
looking forward for new version 👍

**Comment by RandomEngy** [comment]
Fixed in 2.53.
